optimade/server/entry_collections/entry_collections.py

```diff
@@ -293,7 +301,7 @@ def handle_query_params(
         # filter
         if getattr(params, "filter", False):
             cursor_kwargs["filter"] = self.transformer.transform(
-                self.parser.parse(params.filter)
+                self.parser.parse(params.filter)  # type: ignore[union-attr]
```

Comment:
I think it would be cleaner to add "filter" to the SingleEntryQueryParams class. 
We do use the filter to select a single entry based on the id. This may also allow us to make the code for both single and many responses more similar.
But I am not sure how much work it would be. So perhaps this can be done in another PR.

Reply:
It's not too much work, but it will break the OpenAPI schema and will present `filter` as a valid parameter for single entry endpoints (even though we do use the same machinery for it internally). I think its best we leave this one as is; the ignore is just because `mypy` isn't smart enough to resolve dynamic `getattr` guards.